src/libasr/pass/print_arr.cpp

```diff
@@ -122,7 +122,7 @@ class PrintArrVisitor : public PassUtils::PassVisitor<PrintArrVisitor>
             }
             print_stmt = ASRUtils::STMT(ASR::make_Print_t(
                 al, x.base.base.loc, nullptr, body.p, body.size(),
-                nullptr, nullptr));
+                x.m_separator, x.m_end));
```

Comment:
Test with `print(a, b)` where `a` and `b` both are arrays.

Reply:
Yeah I can see that this might fail . Though I am not sure how are arrays expected to be printed . For eg 
```
 (lf) anutosh491@spbhat68:~/lpython/lpython$ cat examples/expr2.py 
def main0():
    x: i32 = 2
    arr1: i32[x]
    arr1[0] = 5
    arr1[1] = 10
    print(arr1)

main0()

(lf) anutosh491@spbhat68:~/lpython/lpython$ ./src/bin/lpython examples/expr2.py
5
10
```
Are the elements expected to be printed out on seperate lines as shown above ?

Reply:
Currently we have the following 
```
(lf) anutosh491@spbhat68:~/lpython/lpython$ cat examples/expr2.py 
def main0():
    x: i32 = 2
    arr1: i32[x]
    arr2: i32[x]
    arr1[0] = 5
    arr1[1] = 10
    arr2[0] = 20
    arr2[1] = 40
    print(arr1, arr2, sep = "abc", end = "xyz")

main0()
# Not implemented yet in LPython:
#if __name__ == "__main__":
#    main()
(lf) anutosh491@spbhat68:~/lpython/lpython$ ./src/bin/lpython examples/expr2.py
5
10

20
40
```

Reply:
Personally I would like to see the output as,

```zsh
5 10
20 40
```

i.e., individual elements of an array should be printed with spaces between them and each array should be printed in a separate line.

Reply:
This the output for something like `print(arr1, arr2)` right ? This tells me the sep and the end keyword by default would be `"\n"`
If we maybe stick to what's default in python , we might expect something like the following . I wouldn't prefer this too !
```
5 1020 40
```

Reply:
I've addressed printing of arrays in my latest comment ( mostly taken 1d arrays into consideration for now) . Here are few examples of printing arrays together and also a random combination at the end 
```
def main0():
    x: i32 = 2
    y: i32 = 4
    z: i32 = 6
    arr1: i32[x]
    arr2: i32[x]
    arr1[0] = 5
    arr1[1] = 10
    arr2[0] = 20
    arr2[1] = 40
    print(arr1, arr2)
    print(arr1, arr2, sep = "abc")
    print(arr1, arr2, end = "xyz\n")
    print(arr1, arr2, sep = "abc", end = "xyz\n")
    print(x, "xyz", arr1, y, z, arr2, "xyz", z, sep = "abc", end = "pqr\n")

main0()

(lf) anutosh491@spbhat68:~/lpython/lpython$ ./src/bin/lpython examples/expr2.py
5 10 
20 40 
5 10abc20 40 
5 10 
20 40xyz
5 10abc20 40xyz
2abcxyzabc5 10abc4abc6abc20 40abcxyzabc6pqr
```

Reply:
@czgdp1807 Could you approve of this output , once done I'll add these print statements in the integration test file I've pushed !

Reply:
```
5 10 
```
extra space at the end of the line? Was there a problem fixing it?

Reply:
What's the CPython output for the same code?

Reply:
Not sure if CPython supports arrays as lpython. 
```
(lf) anutosh491@spbhat68:~/lpython/lpython$ cat examples/expr2.py 
def main0():
    x: i32 = 2
    arr1: i32[x]
    arr2: i32[x]
    arr1[0] = 5
    arr1[1] = 10
    arr2[0] = 20
    arr2[1] = 40
    print(arr1, arr2)

main0()

(lf) anutosh491@spbhat68:~/lpython/lpython$ lpython examples/expr2.py 
5 10 
20 40 

(lf) anutosh491@spbhat68:~/lpython/lpython$ python examples/expr2.py 
Traceback (most recent call last):
  File "/home/anutosh491/lpython/lpython/examples/expr2.py", line 11, in <module>
    main0()
  File "/home/anutosh491/lpython/lpython/examples/expr2.py", line 5, in main0
    arr1[0] = 5
    ^^^^
UnboundLocalError: cannot access local variable 'arr1' where it is not associated with a value
```

Reply:
try it using NumPy arrays

Reply:
See the pattern of how we enable `cpython` with array tests. You should be able to figure out.

Reply:
> try it using NumPy arrays

For something like the following
```
import numpy as np

def main0():
    x: np.int32 = 2
    arr1: np.ndarray[np.int32_t, np.ndim[x]] = np.zeros(x, dtype=np.int32)
    arr2: np.ndarray[np.int32_t, np.ndim[x]] = np.zeros(x, dtype=np.int32)
    arr1[0] = 100
    arr1[1] = 200
    arr2[0] = 300
    arr2[1] = 400
    print(arr1, arr2)
    print(arr1, arr2, sep="abc")
    print(arr1, arr2, end="pqr\n")
    print(arr1, arr2, sep="abc", end="pqr\n")

main0()
```
We get the following result 
```
(lf) anutosh491@spbhat68:~/lpython/lpython$ python examples/expr2.py 
[100 200] [300 400]
[100 200]abc[300 400]
[100 200] [300 400]pqr
[100 200]abc[300 400]pqr
```
This tells me 
1) We might have to use square brackets to enclose the elements .
2) Also default seperator would be a space rather than the end line which I have used here .

Should I go forward with this @Thirumalai-Shaktivel @czgdp1807 ? (Also would it make sense to dedicate a completely new Pr for arrays as I would like to incorporate these changes .... but at the same time , also check for cases other than 1d arrays which currently my changes might not satisfy . The Pr currently would support sep and end keyword with all basic data types and other containers like list/tuple . If we could keep it till this ... i'll make a subsequent pr for arrays !)

Reply:
this works in both CPython and LPython:
```py
from ltypes import i32
from numpy import empty, int32

def main0():
    x: i32 = 2
    arr1: i32[x] = empty(x, dtype=int32)
    arr2: i32[x] = empty(x, dtype=int32)
    arr1[0] = 100
    arr1[1] = 200
    arr2[0] = 300
    arr2[1] = 400
    print(arr1, arr2)
    print(arr1, arr2, sep="abc")
    print(arr1, arr2, end="pqr\n")
    print(arr1, arr2, sep="abc", end="pqr\n")

main0()
```

Reply:
I see. @certik @czgdp1807, what do you think about this?
Should we enclose the array values within SquareBrackets while printing?
CPython output:
```
[100 200] [300 400]
[100 200]abc[300 400]
[100 200] [300 400]pqr
[100 200]abc[300 400]pqr
```

Reply:
I would say enclosing arrays with SquareBrackets isn't a bad addition . If all of us are in .... I'll address it !

Reply:
My lpython output for the following shared program above is 
```
100 200 
300 400 
100 200abc300 400 
100 200 
300 400pqr
100 200abc300 400pqr
```
Basically my default seperator is a newline rather than a space and obviously I haven't includes braces.

Reply:
Yes, we should use square brackets for LPython to be compatible with NumPy, and for LFortran we should not be compatible with other compilers. This can be done in subsequent PRs. Also for both LFortran and LPython having a nice printing of arrays, especially matrices, using unicode, etc. would be awesome, like Julia does it.

Reply:
Thanks for the feedback , I try to cover most of what you've said in a following pr !